src/bp/core/services/telemetry/actions.ts

```diff
@@ -0,0 +1,111 @@
+import { parseActionInstruction } from 'common/action'
+import { BUILTIN_MODULES } from 'common/defaults'
+import LicensingService from 'common/licensing-service'
+import { getSchema } from 'common/telemetry'
+import Database from 'core/database'
+import { calculateHash } from 'core/misc/utils'
+import { TelemetryRepository } from 'core/repositories/telemetry_payload'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import ms from 'ms'
+import path from 'path'
+
+import { GhostService } from '..'
+import { JobService } from '../job-service'
+
+import { TelemetryStats } from './telemetry-stats'
+
+type NextNode = {
+  condition: string
+  node: string
+}
+
+type Node = {
+  id: string
+  name: string
+  next: Array<NextNode>
+  onEnter: Array<string>
+  onReceive: Array<string>
+  type: string
+}
+
+type Flow = {
+  flowName: string
+  botID: string
+  actions: Array<string>
+}
+
+@injectable()
+export class ActionsStats extends TelemetryStats {
+  protected url: string
+  protected lock: string
+  protected interval: number
+
+  constructor(
+    @inject(TYPES.GhostService) ghostService: GhostService,
+    @inject(TYPES.Database) database: Database,
+    @inject(TYPES.LicensingService) licenseService: LicensingService,
+    @inject(TYPES.JobService) jobService: JobService,
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = process.TELEMETRY_URL
+    this.lock = 'botpress:telemetry-actions'
+    this.interval = ms('1d')
+  }
+
+  protected async getStats() {
+    return {
+      ...getSchema(await this.getServerStats(), 'server'),
+      event_type: 'builtin_actions',
+      event_data: { schema: '1.0.0', flows: await this.getFlowsWithActions() }
+    }
+  }
+
+  private async getFlowsWithActions() {
+    const paths = await this.ghostService.bots().directoryListing('/', '*/flows/*.flow.json')
+    let flows
+    try {
+      flows = await Promise.map(paths, async flowPath => {
+        const { dir, base: flowName } = path.parse(flowPath)
+        const botID = dir.split('/')[0]
+        const actions = (await this.ghostService.bots().readFileAsObject<any>(dir, flowName)).nodes
+          .map(node => this.getActionsFromNode(node))
+          .reduce((acc, cur) => [...acc, ...cur])
+        return { flowName, botID, actions }
+      })
+    } catch (error) {
+      return {}
+    }
+    return flows.filter(flow => flow.actions.length > 0).map(flow => this.parseFlow(flow))
+  }
+
+  private getActionsFromNode(node: Node) {
+    const onEnter = node.onEnter ?? []
+    const onReceive = node.onReceive ?? []
+    return [...onEnter, ...onReceive]
```

Comment:
Short n sweet:
```js
const getActionsFromNode => ({onEnter, onReceive}) => [...(onEnter ?? [], ...onReceive  ?? []) 
```

Reply:
fixed in c841929d5213d0ae266b02ec0cf4309e28dd3c68